File: features/offices/src/main/java/com/nivasafinance/features/offices/dto/OfficeCreateRequest.java
package com.nivasafinance.features.offices.dto;

import com.nivasafinance.common.dto.AddressRequest;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OfficeCreateRequest {
    
    @NotNull(message = "Name is required")
    private String name;
    
    @NotNull(message = "Key is required")
    private String key;

    private AddressRequest address;
    
    private Long parentId;
}

File: features/offices/src/main/java/com/nivasafinance/features/offices/dto/OfficeCreateRequest.kt
package com.nivasafinance.features.offices.dto

import com.nivasafinance.features.address.dto.CreateAddressRequest
import java.util.UUID

data class OfficeCreateRequest(
    val name: String,
    val key: String,
    val createAddressRequest: CreateAddressRequest,
    val parentId: UUID? = null
)
